### ISSUE TITLE
phpcs: Unable to locate phpcs. Invalid version string encountered!

### ISSUE THREAD

**Issue by needrunning** [ikappas/vscode-phpcs#150]
mac system

install phpcs via composer global require squizlabs/php_codesniffer

open vscode 

installed  phpcs extention

config it 

exepath 
/Users/usernama/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs

show that

`phpcs: Unable to locate phpcs. Invalid version string encountered!
`

**Comment by Screenfeed** [comment]
Same issue today, while everything was working fine until then.

**Comment by JAW-Dev** [comment]
Yes, I've also been having this issue since the last vscode update.

**Comment by XLearner** [comment]
I use windows system and have the same problem. Have you solved it?

**Comment by needrunning** [comment]
I have solved it via  install PHP CodeSniffe on my system 
![phpcs](https://user-images.githubusercontent.com/3424856/67752516-10feee80-fa6e-11e9-81d3-415024935f93.png)


[https://www.jianshu.com/p/8c16b1f247ed](url)
![phpcs](https://user-images.githubusercontent.com/3424856/67752542-1ceab080-fa6e-11e9-82e9-4b2917ee441c.png)

**Comment by davisben** [comment]
In my case, I'm getting this error because of a PHP deprecation notice that is output when `phpcs --version` is run. The regex expects the output of that command to start with `PHP_CodeSniffer version`, followed by the version number, but the deprecation notice is printed before that, causing the version to not be matched.

**Comment by mjavadhpour** [comment]
I have the same problem. php dump a warning message before any executed command and then the 

```
.\vendor\squizlabs\php_codesniffer\bin\phpcs --version
```

dump this message:

```
PHP Warning:  Version warning: Imagick was compiled against ImageMagick version 1799 but version 1802 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

Warning: Version warning: Imagick was compiled against ImageMagick version 1799 but version 1802 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
PHP_CodeSniffer version 3.5.8 (stable) by Squiz (http://www.squiz.net)
```

I try to change `error_reporting` inside the `php.ini` from `E_ALL` to `E_ERROR`. and then the result of 

```
.\vendor\squizlabs\php_codesniffer\bin\phpcs --version
```

changed to 

```
PHP_CodeSniffer version 3.5.8 (stable) by Squiz (http://www.squiz.net)
```

and my problem was resolved.

@davisben Thank you.

**Comment by CGI-Jay** [comment]
This turned out to be my issue as well - Homebrew had set PHP to 8.4, and an underlying library is throwing out warnings when running `phpcs --version` before the version string.
All my projects run in containers set to PHP 8.3, so I set Homebrew to use PHP 8.3 globally, to prevent the deprecation warnings I'm getting. VSCode is now able to get the correct version when running `phpcs --version` as there are no deprecation warnings before the version ID.

I think the solution is for the VSCode PHPCS plugin to skip passed deprecation warnings when checking the PHPCS version.
